app/lib/backend/model_extensions/service.rb

```diff
@@ -34,11 +33,10 @@ def update_backend_service
       end
 
       def delete_backend_service
-        if account && account.has_bought_cinstance? && !account.api_key.blank?
-          delete_alert_limits(alert_limits)
-          ThreeScale::Core::Service.delete_by_id!(backend_id)
-        end
-
+        has_api_key = account&.has_bought_cinstance? && account.api_key.present?
+        return if account && !has_api_key
```

Comment:
what if `account` is `nil` ? you wouldn't return and carry on, that's expected?

Reply:
@macejmic Yes, it is intentional. If we remove the tenant, then its service it is destroyed, and we want it to be destroyed from Backend too then 😄

Reply:
@macejmic Actually that's the only reason why I changed this method, only to accomplish that 😄

Reply:
It looks like `has_api_key` (better `has_api_key?`) belongs to `Account`.

Reply:
ok! 🍤

Reply:
You can also define a negate method `missing_api_key?` and simplify the safe guard clause.

Reply:
I will see 🤔